File: java/fastcampus-java-spring/part3/chapter04/post/src/main/java/com/example/post/controller/PostApiController.java
package com.example.post.controller;

import com.example.post.dto.PostRequsetDto;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api")
public class PostApiController {
    @PostMapping("/post")
    public void post(@RequestBody PostRequsetDto requestData) {
        System.out.println(requestData);
    }
}

File: spring/inflearn-steven-kotlin-springboot-webmvc/mvc/src/main/kotlin/com/example/mvc/controller/post/PostApiController.kt
package com.example.mvc.controller.post

import com.example.mvc.model.http.UserRequest
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api")
class PostApiController {
    @PostMapping("/post-mapping")
    fun postMapping(): String {
        return "post-mapping"
    }

    @RequestMapping(method = [RequestMethod.POST], path=["/request-mapping"])
    fun requestMapping(): String {
        return "request-mapping"
    }

    @PostMapping("/post-mapping/object")
    fun postMappingObject(@RequestBody userRequest: UserRequest): UserRequest {
        println(userRequest)
        return userRequest
    }
}